src/rgw/rgw_acl.cc

```diff
@@ -90,9 +90,7 @@ void RGWAccessControlList::_add_grant(ACLGrant *grant)
   default:
     {
       rgw_user id;
-      if (!grant->get_id(id)) {
-        ldout(cct, 0) << "ERROR: grant->get_id() failed" << dendl;
```

Comment:
what about this case?

Reply:
`get_id()` only returned false for `ACL_TYPE_GROUP` and `ACL_TYPE_REFERER`, and those were both handled above

the final commit switches to `std::variant` for the grantee type to resolve any ambiguity here